diff -ru LUA53_org/src/loadlib.c LUA53/src/loadlib.c
--- LUA53_org/src/loadlib.c	2014-07-29 09:01:00.000000000 -0700
+++ LUA53/src/loadlib.c	2014-09-29 10:07:55.138880823 -0700
@@ -675,13 +675,27 @@
   if (path == NULL)  /* no environment variable? */
     path = getenv(envname2);  /* try alternative name */
   if (path == NULL || noenv(L))  /* no environment variable? */
+#ifndef _WIN32
+  {
+    lua_getglobal(L, "_LUA_ROOT");
+    luaL_gsub(L, def, LUA_ROOT, luaL_checkstring(L, -1) );
+    lua_remove(L, -2); // remove LUA_ROOT
+  }
+#else
     lua_pushstring(L, def);  /* use default */
+#endif
   else {
     /* replace ";;" by ";AUXMARK;" and then AUXMARK by default path */
     path = luaL_gsub(L, path, LUA_PATH_SEP LUA_PATH_SEP,
                               LUA_PATH_SEP AUXMARK LUA_PATH_SEP);
-    luaL_gsub(L, path, AUXMARK, def);
-    lua_remove(L, -2);
+    path = luaL_gsub(L, path, AUXMARK, def);
+#ifndef _WIN32
+    lua_getglobal(L, "LUA_ROOT");
+    luaL_gsub(L, path, LUA_ROOT, luaL_checkstring(L, -1) );
+    lua_remove(L, -2); // remove LUA_ROOT
+    lua_remove(L, -2); // remove AUXMARK->def path
+#endif
+    lua_remove(L, -2); // remove ;; -> AUXMARK path
   }
   setprogdir(L);
   lua_setfield(L, -2, fieldname);
diff -ru LUA53_org/src/lua.c LUA53/src/lua.c
--- LUA53_org/src/lua.c	2014-06-30 12:48:08.000000000 -0700
+++ LUA53/src/lua.c	2014-09-29 10:12:40.479960849 -0700
@@ -13,6 +13,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <libgen.h>   //dirname()
 
 #include "lua.h"
 
@@ -567,6 +568,10 @@
     lua_pushboolean(L, 1);  /* signal for libraries to ignore env. vars. */
     lua_setfield(L, LUA_REGISTRYINDEX, "LUA_NOENV");
   }
+#ifndef _WIN32
+  lua_pushfstring(L, "%s/", dirname(dirname(argv[0])));
+  lua_setglobal(L, "_LUA_ROOT");
+#endif
   luaL_openlibs(L);  /* open standard libraries */
   createargtable(L, argv, argc, script);  /* create table 'arg' */
   if (!(args & has_E)) {  /* no option '-E'? */
diff -ru LUA53_org/src/luaconf.h LUA53/src/luaconf.h
--- LUA53_org/src/luaconf.h	2014-07-24 12:33:29.000000000 -0700
+++ LUA53/src/luaconf.h	2014-09-29 10:15:00.715459099 -0700
@@ -175,7 +175,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	".../"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
