diff -ru lua-5.2.1/src/luaconf.h lua-5.2.1.new/src/luaconf.h
--- lua-5.2.1/src/luaconf.h	2012-05-11 16:14:42.000000000 +0200
+++ lua-5.2.1.new/src/luaconf.h	2012-09-12 22:40:27.986622772 +0200
@@ -100,7 +100,7 @@
 #else			/* }{ */
 
 #define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	L_XT_ROOT
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
diff -ru lua-5.2.1/src/linit.c lua-5.2.1.new/src/linit.c
--- lua-5.2.1/src/linit.c	2012-03-09 17:32:16.000000000 +0100
+++ lua-5.2.1.new/src/linit.c	2012-09-12 22:38:08.591386896 +0200
@@ -21,6 +21,7 @@
 #include "lualib.h"
 #include "lauxlib.h"
 
+#include "l_xt.h"
 
 /*
 ** these libs are loaded by lua.c and are readily available to any Lua
@@ -45,6 +45,7 @@
 ** these libs are preloaded and must be required before used
 */
 static const luaL_Reg preloadedlibs[] = {
+  {"xt", luaopen_xt},
   {NULL, NULL}
 };
 
